erpnext/selling/page/point_of_sale/point_of_sale.py

```diff
@@ -88,11 +88,15 @@ def get_conditions(item_code, serial_no, batch_no, barcode):
 	return '%%%s%%'%(frappe.db.escape(item_code)), condition
 
 @frappe.whitelist()
-def submit_invoice(doc):
+def submit_invoice(doc,islocal_dummy):
 	if isinstance(doc, basestring):
 		args = json.loads(doc)
 
-	doc = frappe.new_doc('Sales Invoice')
+	if islocal_dummy:
+		doc = frappe.get_doc('Sales Invoice',args["name"])
```

Comment:
Looks like this is causing problems with saving
![peek 2018-01-12 12-54](https://user-images.githubusercontent.com/818803/34879646-3548f5ac-f7ae-11e7-8075-5da829fc715a.gif)

Reply:
@tundebabzy  Fixed the bug and refactored the code a bit.

![jmi_pr_1_rev](https://user-images.githubusercontent.com/25682214/34893839-f8d8408a-f804-11e7-80ec-4073576935f7.gif)